src/forms/inputcounter/InputCounter.tsx

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright (c) 2018 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+import * as React from "react";
+import * as utils from "../../utils";
+import {Icon} from "../../icon";
+import {Input} from "../input/Input";
+import "./InputCounter.css";
+
+type InputCounterProps = {
+    className?: string;
+    maxValue?: number;
+    minValue?: number;
+    defaultValue?: number;
+    errMsg?: string;
+    width?: string;
+    onChange?: (evt: React.ChangeEvent<HTMLInputElement>) => void;
+};
+
+type InputCounterState = {
+    value: number;
+    errorMsg: string;
+};
+
+export class InputCounter extends React.PureComponent<InputCounterProps, InputCounterState> {
+    public static defaultProps: Partial<InputCounterProps> = {
+        minValue: Number.MIN_SAFE_INTEGER,
+        maxValue: Number.MAX_SAFE_INTEGER,
+        errMsg: "Value should be",
+        width: "6rem",
+    };
+    constructor(props: any) {
+        super(props);
+
+        //Default Value should be >= to minValue
+        let defaultValue = isNaN(props.defaultValue) ? 0 : Math.max(props.defaultValue, props.minValue);
+        //Default Value should be <= to minValue
+        defaultValue = Math.min(defaultValue, props.maxValue);
+        this.state = {
+            value: defaultValue,
+            errorMsg: "",
+        };
+    }
+
+    private handleChange = (evt: React.ChangeEvent<HTMLInputElement>) => {
+        if (isNaN(parseInt(evt.target.value))) {
+            this.setState({
+                value: isNaN(this.props.minValue!) ? 0 : Math.max(this.props.defaultValue, this.props.minValue!),
+            });
+        } else if (parseInt(evt.target.value) > this.props.maxValue!) {
+            let maxValueError = this.props.errMsg + " <= " + this.props.maxValue;
+            this.setState({errorMsg: maxValueError});
+            this.setState({value: parseInt(evt.target.value)});
+        } else if (parseInt(evt.target.value) < this.props.minValue!) {
+            let minValueError = this.props.errMsg + " >= " + this.props.minValue;
+            this.setState({errorMsg: minValueError});
+            this.setState({value: parseInt(evt.target.value)});
+        } else {
+            this.setState({value: parseInt(evt.target.value)});
+            this.setState({errorMsg: ""});
+        }
+        const {onChange} = this.props;
+        if (onChange) onChange(evt);
+    };
+
+    private handleIncrement() {
+        if (this.state.value < this.props.maxValue!) {
```

Comment:
Same logic is repeated in handleChange() to check validity of value before increment/decrements. Can we make common function of it and reuse at both places ?

Reply:
Login is different in both the places. In handleIncrement(), it is checking against maxValue and in handleDecrement(). it is checking against minValue.

Reply:
Yes I know.  I just combined 2 comments here. We can create 2 common methods and  reuse them.